dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -92,107 +94,129 @@ public boolean processDailyFiles() {
+        } catch (JAXBException e) {
+            failedContributions.put(concorContributionId, e.getClass().getName() + ": " + e.getMessage());
+            log.error("Failed to unmarshal contribution data XML, concorContributionId = {}", concorContributionId, e);
+            eventService.logConcor(concorContributionId, FETCHED_FROM_MAAT, batchId, null, INTERNAL_SERVER_ERROR, "Failed to unmarshal contribution data XML");
+            return null;
+        }
+
+        if (feature.outgoingAnonymized()) {
+            // anonymize the data when flag is true - only for non production environments
+            currentContribution = anonymisingDataService.anonymise(currentContribution);
+        }
+        eventService.logConcor(concorContributionId, FETCHED_FROM_MAAT, batchId, currentContribution, OK, null);
+        return currentContribution;
+    }
+
+    // External Call Executions Methods
+
+    @Retry(name = SERVICE_NAME)
+    private int executeContributionProcessedAckCall(ContributionProcessedRequest contributionProcessedRequest) {
+        int result;
+        if (!feature.incomingIsolated()) {
+            result = contributionClient.sendLogContributionProcessed(contributionProcessedRequest);
+        } else {
+            log.info("processContributionUpdate: not calling MAAT API sendLogContributionProcessed() because `feature.incoming-isolated=true`");
+            result = 0; // avoid updating MAAT DB.
+        }
+        logContributionAsyncEvent(contributionProcessedRequest, OK);
+        return result;
+    }
+
+    @Retry(name = SERVICE_NAME)
+    private List<ConcorContribEntry> executeGetContributionsCall(int startingId) {
+        List<ConcorContribEntry> contributionsList = contributionClient.getContributions(ContributionRecordStatus.ACTIVE.name(), startingId, getContributionBatchSize);
+        eventService.logConcor(null, FETCHED_FROM_MAAT, batchId, null, OK, String.format("Fetched:%s",contributionsList.size()));
+        return contributionsList;
+    }
+
     @Retry(name = SERVICE_NAME)
-    public Integer contributionUpdateRequest(String xmlContent, List<String> concorContributionIdList, int numberOfRecords, String fileName, String fileAckXML) throws HttpServerErrorException {
+    private void executeSendConcorToDrcCall(BigInteger concorContributionId, CONTRIBUTIONS currentContribution, Map<BigInteger, String> failedContributions) {
+        final var request = ConcorContributionReqForDrc.of(concorContributionId.intValue(), currentContribution);
+        if (!feature.outgoingIsolated()) {
+            drcClient.sendConcorContributionReqToDrc(request);
+            log.info("Sent contribution data to DRC, concorContributionId = {}, maatId = {}", concorContributionId, currentContribution.getMaatId());
+        } else {
+            log.info("Skipping contribution data to DRC, concorContributionId = {}, maatId = {}", concorContributionId, currentContribution.getMaatId());
```

Comment:
Wherever any processing is skipped because of a feature flag setting, can the word "feature" be included somehow in the log message? This is helpful when gathering evidence that the flags are working, by filtering log messages that have the word "feature" in them.